linetools/spectra/xspectrum1d.py

```diff
@@ -84,92 +75,205 @@ def from_file(self, ifile, **kwargs):
           Filename
         """
         # put import here to avoid circular import with io.py
-        from .io import readspec
-        slf = readspec(ifile, **kwargs)
+        #from .io import readspec
+        from linetools.spectra import io as tio
+        reload(tio)
+        slf = tio.readspec(ifile, **kwargs)
         return slf
 
     @classmethod
     def from_spec1d(cls, spec1d):
         """ Input Spectrum1D
-        """
-        # Giddy up
-        slf = cls(flux=spec1d.flux, wcs=spec1d.wcs, unit=spec1d.unit,
-                  uncertainty=spec1d.uncertainty, mask=spec1d.mask,
-                  meta=spec1d.meta.copy())
-        slf.co = (spec1d.co if hasattr(spec1d, 'co') else None)
 
+        Avoids error array for now
+        """
+        slf = cls.from_tuple((spec1d.dispersion, spec1d.flux))
+        #slf = cls(flux=spec1d.flux, wcs=spec1d.wcs, unit=spec1d.unit,
+        #          uncertainty=spec1d.uncertainty, mask=spec1d.mask,
+        #          meta=spec1d.meta.copy())
         return slf
 
     @classmethod
     def from_tuple(cls, ituple, sort=True, **kwargs):
-        """Make an XSpectrum1D from a tuple of arrays.
+        """Make an XSpectrum1D from a tuple of numpy arrays or Quantity arrays
 
         Parameters
         ----------
         ituple : (wave,flux), (wave,flux,sig) or (wave,flux,sig,cont)
+          ndarray or Quantity array
           If wave is unitless, Angstroms are assumed
+          If flux is unitless, it is made dimensionless
```

Comment:
Mention that the units (if any) of sig and cont are assumed to be those of flux (?)